internal/pkg/agent/application/upgrade/rollback.go

```diff
@@ -143,12 +143,32 @@ func restartAgent(ctx context.Context) error {
 		return nil
 	}
 
+	restartViaServiceFn := func(ctx context.Context) error {
+		topPath := paths.Top()
+		install.StopService(topPath)
+		err := install.StartService(topPath)
```

Comment:
Would it better to create a new `install.RestartServive(topPath)` wrapper around these two calls?

Reply:
Added `install.RestartService` in 6cb1746c7.  I'm taking care of adding logging to the restart step in #3245.